src/land/heightmap.rs

```diff
@@ -36,21 +36,21 @@ impl<'a> TextureHeightmap<'a> {
 
         Ok(TextureHeightmap {
             texture,
-            size: (texture.size.x() as u32 - 2, texture.size.y() as u32 - 2),
+            size: (texture.size.x() as u16 - 2, texture.size.y() as u16 - 2),
```

Comment:
I'd still prefer if this both utilized `From` and the `size` field in general was some form of type-safe newtype wrapper ensuring correct usage.

Reply:
- `TextureHeightmap::size` is not part of the public API.
 - Unfortunately the Bevy `size` field is _totally_ the wrong type (`f32` to specify texture size). I don't think there's a much better way to do the conversion here (maybe try_into and expect)